code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -364,13 +364,22 @@
 /proc/hasHUD(mob/M as mob, hudtype)
 	if(istype(M, /mob/living/carbon/human))
 		var/mob/living/carbon/human/H = M
-		switch(hudtype)
-			if("security")
-				return istype(H.glasses, /obj/item/clothing/glasses/hud/security) || istype(H.glasses, /obj/item/clothing/glasses/sunglasses/sechud)
-			if("medical")
-				return istype(H.glasses, /obj/item/clothing/glasses/hud/health)
-			else
-				return 0
+		var/obj/item/clothing/glasses/G = H.glasses
+		var/eye_hud = 0
+		var/hud = 0
+		// Checks for eye sensor HUD
+		var/obj/item/organ/internal/augment/eye_sensors/E = locate() in H.internal_organs
+		if(E)
+			eye_hud = E.check_hud(hudtype)
+		if(G)
+			switch(hudtype)
```

Comment:
Would be preferable if the strings "security" and "medical" were defines, if you want to do that that'd be cool.